src/block_pos.rs

```diff
@@ -88,6 +88,19 @@ impl From<BlockPos> for Vec3<i32> {
     }
 }
 
+/// Only [`Into`] is implemented, as [`f64`] can represent all [`i32`] values,
+/// but not the other way around
+impl Into<Vec3<f64>> for BlockPos {
+    fn into(self) -> Vec3<f64> {
+        let BlockPos { x, y, z } = self;
+        Vec3 {
+            x: x as f64,
+            y: y as f64,
+            z: z as f64,
+        }
+    }
+}
+
```

Comment:
You should never `impl Into` in Rust. Use `From` instead. How about
```rust
impl From<BlockPos> for Vec3<f64> {
    fn from(pos: BlockPos) -> Self {
        Vec3::new(pos.x.into(), pos.y.into(), pos.z.into())
    }
}
```

Reply:
Yeah I misunderstood the blanket `impl`, I wanted to avoid lossy conversions from `f64` to `i32`. Anyway, I think it may not be necessary, so I will undo it.